src/modules/onchain-logic/views/Dashboard.tsx

```diff
@@ -0,0 +1,158 @@
+// Copyright © 2021 Kaleido, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+  CircularProgress,
+  Grid,
+  IconButtonProps,
+  TablePagination,
+  Typography,
+} from '@mui/material';
+import React, { useState, useContext } from 'react';
+import { IDataTableRecord, IContractInterface } from '../../../core/interfaces';
+import { useQuery } from 'react-query';
+import { fetchCatcher } from '../../../core/utils';
+import { useOnChainLogicTranslation } from '../registration';
+import { DataTable } from '../../../core/components/DataTable/DataTable';
+import { NamespaceContext } from '../../../core/contexts/NamespaceContext';
+import FileCodeOutlineIcon from 'mdi-react/FileCodeOutlineIcon';
+import { DataTableEmptyState } from '../../../core/components/DataTable/DataTableEmptyState';
+
+const PAGE_LIMITS = [10, 25];
+
+export const Dashboard: () => JSX.Element = () => {
+  const { t } = useOnChainLogicTranslation();
+
+  const [currentPage, setCurrentPage] = useState(0);
+  const [rowsPerPage, setRowsPerPage] = useState(PAGE_LIMITS[0]);
+  const { selectedNamespace } = useContext(NamespaceContext);
+
+  const getContractsInterfacesApi = `/api/v1/namespaces/${selectedNamespace}/contracts/interfaces?limit=${rowsPerPage}&skip=${
+    rowsPerPage * currentPage
+  }`;
+  const getContractsInterfaces = () => fetchCatcher(getContractsInterfacesApi);
+  const { data: contractInterfaces, isLoading: contractInterfacesLoading } =
+    useQuery<IContractInterface[]>(
+      getContractsInterfacesApi,
+      getContractsInterfaces
+    );
+
+  const columnHeaders = [
+    t('contract'),
+    t('id'),
+    t('message'),
+    t('description'),
+    t('version'),
```

Comment:
I would probably put "name" and "version" first in this list. Not sure if "message" is terribly useful at this level (more useful in a detail view where you can provide a clickthrough to see the message) - but if we want to fill a column then it's fine to show it.